libs/testing/cypress/node/src/lib/plugins/db.ts

```diff
@@ -47,39 +55,67 @@ export async function validateOrg(orgName: string) {
   const permissions = createPermissions(permissionsRef.data());
-  const adminIds = Object.keys(permissions.roles).filter(userId => {
-    return (
-      permissions.roles[userId] === 'superAdmin' ||
-      permissions.roles[userId] === 'admin'
-    );
-  });
-  const randomIndex = Math.floor(Math.random() * adminIds.length);
-  const adminId = adminIds[randomIndex];
-  const adminRef = await db.doc(`users/${adminId}`).get();
-  return createUser(adminRef.data());
+  const userIds = Object.keys(permissions.roles).filter(uid =>
+    type==='member' ? permissions.roles[uid] === 'member' : permissions.roles[uid] === 'superAdmin' || 'admin');
+  const randomIndex = Math.floor(Math.random() * userIds.length);
+  const userId = userIds[randomIndex];
+  const userRef = await db.doc(`users/${userId}`).get();
+  return createUser(userRef.data());
 }
 
-export function deleteUser(userId: string) {
-  return db.doc(`users/${userId}`).delete();
+//* ORGS + MOVIES
+
+async function getOrgMovies(orgId: string) {
+  const movies = [];
+  const userQuery = await db.collection('movies').where('orgIds', 'array-contains', orgId).get();
+  userQuery.forEach(doc => movies.push(doc.data()));
+  return movies.filter(movie => movie.title.international || movie.title.original);
 }
 
-export async function getOrgByName(orgName: string) {
-  const userQuery = await db.collection('orgs').where('denomination.full', '==', orgName).get();
-  const [org] = userQuery.docs;
-  if(!org) return null;
-  return createOrganization(org.data());
+//* ORGS + USERS + MOVIES
+
+export async function getRandomScreeningData(data: { app: App, access: ModuleAccess, userType: 'admin' | 'member', moviesWithScreener?: boolean}) {
+  const { app, access, userType, moviesWithScreener } = data;
+  let result = {};
+  do {
+    const org = await getRandomOrg({app, access, userType});
+    const user = await getRandomOrgUser(org.id, userType);
+    let movies = (await getOrgMovies(org.id))
+      .filter(movie => movie.title.international && movie.app.festival.status == "accepted" && movie.app.festival.access == true);
+    if(!isUndefined(moviesWithScreener)) movies = movies.filter(movie => moviesWithScreener ? movie.promotional.videos.screener.jwPlayerId : !movie.promotional.videos.screener.jwPlayerId);
+    if (movies.length) result = { org, user, movies };
+  } while (!Object.keys(result).length);
+  return result;
 }
 
-export function deleteOrg(orgId: string) {
-  return db.doc(`orgs/${orgId}`).delete();
-}
\ No newline at end of file
+export async function getRandomScreeningDataArray(data: { app: App; access: ModuleAccess, userType: 'admin' | 'member', number: number, moviesWithScreener: boolean}) {
+  const { app, access, userType, number, moviesWithScreener } = data;
+  const result= [];
+  for( let i = 0; i < number; i++) {
+    result.push(await getRandomScreeningData({app, access, userType, moviesWithScreener}));
+  }
+  return result;
+}
```

Comment:
```suggestion
export  function getRandomScreeningDataArray(data: { app: App, access: ModuleAccess, userType: UserRole, number: number, moviesWithScreener: boolean}) {
  const { app, access, userType, number, moviesWithScreener } = data;
  const promises= [];
  for( let i = 0; i < number; i++) {
    promises.push(getRandomScreeningData({app, access, userType, moviesWithScreener}));
  }
  return promise.all(promises);
}
```

But are you using this function ?

Reply:
not yet, i'll delete it if I really don't use it when the test is finished 👌